### PR TITLE
Query DB optims

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:

- Query ParityDB first
- Read-only CAR files are sorted by heaviest tipset

## Reference issue to close (if applicable)

Closes

## Other information and links

To exercise some load, I've imported 44 diff calibnet car files and done a simple export.

`forest-cli snapshot export -t 1395675 -d 900`

| branch | elapsed time |
| - | - |
| main | 92s |
| db-query-optim | 60s |

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/db/car/many.rs [hunk 1]
```diff
@@ -64,14 +65,29 @@ impl<WriterT> ManyCar<WriterT> {
...
+        files: impl Iterator<Item = PathBuf>,
+    ) -> anyhow::Result<Self> {
         self.read_only_files(files)?;
         Ok(self)
     }
 
-    pub fn read_only_files(&self, files: impl Iterator<Item = PathBuf>) -> io::Result<()> {
+    pub fn read_only_files(&self, files: impl Iterator<Item = PathBuf>) -> anyhow::Result<()> {
+        let mut cars = vec![];
         for file in files {
```
Comment (hanabi1224) [inline src/db/car/many.rs:78]: This does not sort the existing CARs though. How about refactoring `read_only` field using [`BinaryHeap`](https://doc.rust-lang.org/stable/std/collections/struct.BinaryHeap.html)

Comment (elmattic) [reply src/db/car/many.rs:78]: Good idea.

File: src/db/car/many.rs [hunk 2]
```diff
@@ -49,40 +88,53 @@ impl<WriterT> ManyCar<WriterT> {
...
+
+            self.read_only(car)?;
         }
 
         Ok(())
     }
 
+    // TODO: update
     pub fn heaviest_tipset(&self) -> anyhow::Result<Tipset> {
         let tipsets = self
```
Comment (hanabi1224) [inline src/db/car/many.rs:133]: Should it be sufficient to load heaviest from the first CAR now?

Comment (elmattic) [reply src/db/car/many.rs]: Yes, I'm using `peek` method.

**Review by hanabi1224 — APPROVED** [review]

**Review by sudo-shashank — APPROVED** [review]

**Pull request merged**